lib/jobsub_api.py

```diff
@@ -0,0 +1,62 @@
+from typing import List, Optional, Generator
+import re
+import sys
+import contextlib
+from io import StringIO
+from mains import jobsub_submit_main, jobsub_fetchlog_main, jobsub_cmd_main
+
+
+@contextlib.contextmanager
+def output_saver(should_i: bool) -> Generator[StringIO, StringIO, StringIO]:
```

Comment:
Perhaps I'm missing something, but shouldn't this type expression be `Generator[StringIO, None, StringIO]`, since we're not sending anything to the generator?

Reply:
Hm.. well PEP 484 says: "The return type of generator functions can be annotated by the generic type Generator[yield_type, send_type, return_type] provided by typing.py module:"  so I think that means
it should be Generator[StringIO, bool, StringIO]  because we pass a boolean in.

Reply:
Well, that PEP mentions the `send_type`, which I don't believe applies here.  The `send_type` as I understand it is only if you're sending a value to the generator, which we're not - we're instantiating the generator with a bool that affects its behavior.

If we look at https://peps.python.org/pep-0484/#annotating-generator-functions-and-coroutines, it shows an example, where they're sending values to the generator like 

```python
res = yield round(res)
```

In that example, they're passing floats to the generator, which rounds them and then yields the rounded value back to the caller, waiting for the next value to round.

I don't believe we're doing that here.